__tests__/client/service-worker/index.spec.js

```diff
@@ -84,6 +90,22 @@ describe('initializeServiceWorker', () => {
     expect(serviceWorkerClient).not.toHaveBeenCalled();
   });
 
+  test('when serviceWorker is disabled, clears the cache of any entries', async () => {
+    expect.assertions(5);
+
+    await (await caches.open('one-app-test')).put(new Request('/index.js'), null);
+
+    expect(caches.snapshot()).toEqual({
+      'one-app-test': {
+        'http://localhost/index.js': null,
+      },
+    });
+    await expect(initializeServiceWorker({ serviceWorker: false })).resolves.toBe(registration);
+    expect(registration.unregister).toHaveBeenCalledTimes(1);
+    expect(serviceWorkerClient).not.toHaveBeenCalled();
+    expect(caches.snapshot()).toEqual({});
```

Comment:
is it worth to add a tests that deletes only the keys cached by the service worker and nothing else, as I understand it, it should keep keys in the cache that don't have anything to do with our SW

Reply:
That's a good point. I'll update to add this.

Reply:
I have updated to filter out cache keys created by the service worker with the prefix `__sw` ( 2472120 )